kahuna/public/js/components/gr-collections-panel/gr-collections-panel.js

```diff
@@ -30,22 +35,136 @@ grCollectionsPanel.controller('GrCollectionsPanelCtrl', [
         ctrl.error = true;
     }).catch(() => ctrl.error = true);
 
+    ctrl.selectedImages$ = selectedImages$;
 }]);
 
-grCollectionsPanel.controller('GrNodeCtrl', ['collections', function(collections) {
+grCollectionsPanel.factory('collectionsTreeState', ['$window', function($window) {
+    const localStorageKey = 'collectionsTreeState';
+    const defaultShowState = true;
+
+    function getStateFromStorage() {
+        const jsonStr = $window.localStorage.getItem(localStorageKey) || '{}';
+        return JSON.parse(jsonStr);
+    }
+
+
+    function setState(pathId, show) {
+        const newState = angular.extend({}, getStateFromStorage(), {[pathId]: show});
```

Comment:
As discussed, should we only store one of the states (either all `true` or all `false` values), and let the rest be the default?